frostsnapp/lib/stream_ext.dart

```diff
@@ -12,12 +12,23 @@ extension StreamToBehaviorSubjectExtension<T> on Stream<T> {
         : BehaviorSubject<T>();
 
     // Listen to the original stream and forward events to the BehaviorSubject
-    listen(
+    // CRITICAL FIX: Store subscription so it can be cancelled
+    late StreamSubscription<T> subscription;
+    subscription = listen(
       (data) => subject.add(data),
       onError: (error) => subject.addError(error),
-      onDone: () => subject.close(),
+      onDone: () {
+        subject.close();
+        // No need to cancel subscription here as stream is done
+      },
     );
 
+    // CRITICAL FIX: Cancel subscription when subject is closed manually
+    // This prevents memory leaks when the subject is disposed before stream completes
+    subject.onCancel = () {
```

Comment:
Was this a real problem? Maybe plugging in and unplugging devices over and over would take up lots of memory? My understanding is that this stream is short-lived and would be freed by simply closing the dialog.

I'm probably over considering due to "CRITICAL".